app/webpacker/controllers/dependent_select_controller.js

```diff
@@ -29,8 +29,14 @@ export default class extends Controller {
       this.addOption(item[0], item[1]);
     });
 
-    this.selectTarget.tomselect?.sync();
-    this.selectTarget.tomselect?.addItem(options[0]?.[1]);
+    if (options.length == 0) {
+      this.selectTarget.tomselect?.disable();
+    } else {
+      this.selectTarget.tomselect?.enable();
+      this.selectTarget.tomselect?.addItem(options[0]?.[1]);
+      this.selectTarget.tomselect?.sync();
+      this.selectTarget.tomselect?.setValue(null);
+    }
```

Comment:
All of these statements are based on the same condition, using the "optional chaining" operator (`?.`, which I only just discovered exists and is similar to Ruby's `&.`).
So this code could be simplified a lot if we just check the presence of `tomselect` once. Given that this function doesn't do anything else afterwards, we can simply return early. IE:

```suggestion
    const tomselect = this.selectTarget.tomselect
    if (!tomselect) { return; }
   
    if (options.length == 0) {
      tomselect.disable();
    } else {
      tomselect.enable();
      tomselect.addItem(options[0]?.[1]);
      tomselect.sync();
      tomselect.setValue(null);
    }
```

Reply:
Good idea, fixed by https://github.com/openfoodfoundation/openfoodnetwork/pull/10614/commits/9f508787b8bbf558d8b89e4dfebab37163232933

Reply:
Oops, I missed this a minute ago. I'll just point out that my suggested early `return` method is to avoid code branching on if/else, and indentation. This adds a bit of complexity to the code and can make it harder to spot mistakes, so that's why return early is preferred.

This is a relevant reference for Ruby: https://github.com/rubocop/ruby-style-guide#no-nested-conditionals

I'm not sure how common this is in Javascript though, so am happy to leave this as-is for now.